### PR TITLE
Feat: [OR-166] Switch rebrand

### PR DESCRIPTION
**Pull request by vicky-comeau**
## Summary

Applied the new look to the switch component.

### CODE REVIEW

File: packages/components/src/switch/src/Switch.css [hunk 1]
```diff
@@ -1,38 +1,50 @@
 .o-ui-switch {
+    --o-ui-switch-hover-background: var(--hop-neutral-surface-hover);
```
Comment (fraincs) [inline packages/components/src/switch/src/Switch.css:2]: I would rename these to follow Hopper tokens naming convention: 

```
--o-ui-switch-background-hover: var(--hop-neutral-surface-hover);
--o-ui-switch-border-hover: var(--hop-neutral-border-hover);
--o-ui-switch-icon-hover: var(--hop-neutral-icon-hover);
--o-ui-switch-background-focus: var(--hop-neutral-surface-hover);
--o-ui-switch-border-focus: var(--hop-neutral-border-hover);
--o-ui-switch-icon-focus: var(--hop-neutral-icon-hover);
```

File: packages/components/src/switch/src/Switch.css [hunk 2]
```diff
@@ -1,38 +1,50 @@
...
+    --o-ui-switch-hover-border: var(--hop-neutral-border-hover);
+    --o-ui-switch-hover-icon: var(--hop-neutral-icon-hover);
+    --o-ui-switch-focus-background: var(--hop-neutral-surface-hover);
+    --o-ui-switch-focus-border: var(--hop-neutral-border-hover);
+    --o-ui-switch-focus-icon: var(--hop-neutral-icon-hover);
```
Comment (fraincs) [inline packages/components/src/switch/src/Switch.css:7]: --o-ui-switch-focus-icon does not indicates what it'll affect

File: packages/components/src/switch/src/Switch.css [hunk 3]
```diff
@@ -42,103 +54,158 @@
...
+.o-ui-switch-disabled .o-ui-switch-control {
+    background-color: var(--hop-neutral-surface-disabled);
+    border-color: var(--hop-neutral-border-disabled);
+}
+
+.o-ui-switch-disabled .o-ui-switch-control::before {
+    background-color: var(--hop-neutral-icon-disabled);
+}
+
+/* INVALID */
```
Comment (fraincs) [inline packages/components/src/switch/src/Switch.css:138]: The design has no notion of invalid, I don't see how a toggle could be in error. I would remove this until we hear back from design, it they need it, we could do another PR to add this.

**Comment by fraincs** [comment]
image
The small Switch should be 16 px high.

image
The medium Switch should be 24px high.

The border seems to be the issue, you could use an outline as it won't affect the element height.

File: packages/components/src/switch/src/Switch.css [hunk 4]
```diff
@@ -1,38 +1,50 @@
...
 }
 
 /* SWITCH */
-.o-ui-switch .o-ui-switch-control {
+.o-ui-switch-control {
     display: inline-block;
-    background-color: var(--o-ui-bg-alias-hard-break);
-    border-radius: var(--o-ui-br-pill);
+    background-color: var(--hop-neutral-surface);
+    border: 0.06rem solid var(--hop-neutral-border);
```
[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/components/src/switch/src/Switch.css:25]: My bad, that's all I saw in figma. I put it to 0.625rem. Keep in mind, most browsers will show it as 0.8px, even if you put 1px:
image
It's annoying.

File: packages/components/src/switch/src/Switch.css [hunk 5]
```diff
@@ -1,38 +1,50 @@
...
 }
 
-.o-ui-switch .o-ui-switch-control::before {
+.o-ui-switch-control::before {
+    --o-ui-switch-control-before-transform-translate-value: var(--hop-space-inline-xs);
     content: "";
     display: inline-block;
-    background-color: var(--o-ui-bg-alias-static-white);
-    border-radius: var(--o-ui-br-circular);
+    background-color: var(--hop-neutral-icon);
```
[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/components/src/switch/src/Switch.css:37]: but it actually uses icon in the design so I tried to be consistent:
image
Probably because they are using an icon.

[the remark this answers is not in the document]
Comment (fraincs) [reply packages/components/src/switch/src/Switch.css:37]: I'm splitting hairs here for sure. To me it's not an icon as it's not a SVG, Design had no way to tell how we would implement this. I'm just concerned that one day we have an icon in the circle(we had it in Orbit once).

[the remark this answers is not in the document]
Comment (vicky-comeau) [reply packages/components/src/switch/src/Switch.css:37]: Well considering --hop-neutral-surface is white, unless I go with the strong version, I'm going to stick with icon. It's what the figma has, it can still be considered a sort of icon. If ever we add an icon or change this somehow, it'll probably be reflected in the design anyway. We can fix it if it comes up.

**Review by fraincs — APPROVED** [review]
